src/Operators/spacings_and_areas_and_volumes.jl

```diff
@@ -47,19 +47,57 @@ The operators in this file fall into three categories:
 @inline Δzᵃᵃᶜ(i, j, k, grid::VerticallyStretchedRectilinearGrid) = @inbounds grid.Δzᵃᵃᶜ[k]
 
 #####
-##### "Spacings" in Flat directions. Here we dispatch to `one`. This abuse of notation
-##### makes volumes correct as we want to multiply by 1, and avoids issues with
-##### derivatives such as those involved in the pressure correction step.
+##### "Spacings" in Flat directions for rectilinear grids.
+##### Here we dispatch all spacings to `one`. This abuse of notation
+##### makes volumes and areas correct.
+#####
+##### Note: Vertical metrics are specific to each rectilinear grid type; therefore
+##### we must dispatch on Flat vertical directions for each grid independently.
 #####
 
 using Oceananigans.Grids: Flat
 
-@inline Δx(   i, j, k, grid::RegularRectilinearGrid{FT, Flat})         where FT           = one(FT)
-@inline Δy(   i, j, k, grid::RegularRectilinearGrid{FT, TX, Flat})     where {FT, TX}     = one(FT)
-@inline ΔzC(  i, j, k, grid::RegularRectilinearGrid{FT, TX, TY, Flat}) where {FT, TX, TY} = one(FT)
-@inline ΔzF(  i, j, k, grid::RegularRectilinearGrid{FT, TX, TY, Flat}) where {FT, TX, TY} = one(FT)
-@inline Δzᵃᵃᶠ(i, j, k, grid::RegularRectilinearGrid{FT, TX, TY, Flat}) where {FT, TX, TY} = one(FT)
-@inline Δzᵃᵃᶜ(i, j, k, grid::RegularRectilinearGrid{FT, TX, TY, Flat}) where {FT, TX, TY} = one(FT)
+#####
+##### Horizontal metrics for AbstractRectilinearGrid
+#####
+
+const XFlatARG = AbstractRectilinearGrid{<:Any, <:Flat}
+const YFlatARG = AbstractRectilinearGrid{<:Any, <:Any, <:Flat}
+
+@inline Δx(i, j, k, grid::XFlatARG) = one(eltype(grid))
```

Comment:
Why when, e.g., `x` is Flat then `Δx=1`? Physically this should be 0. I guess it's a "hack" to allow for other things to work? If so perhaps a comment above with a small clarification?

Reply:
The comment refers to it as an "abuse of notation":

```
#####
##### "Spacings" in Flat directions for rectilinear grids.
##### Here we dispatch all spacings to `one`. This abuse of notation
##### makes volumes and areas correct.
#####
##### Note: Vertical metrics are specific to each rectilinear grid type; therefore
##### we must dispatch on Flat vertical directions for each grid independently.
#####
```

Is this clarifying enough or perhaps there should be more?

More specifically: for rectilinear grids, the volumes and areas are products of the linear spacings (dx, dy, dz). When a dimension is `Flat` I think we want to interpret that dimension as "dropped", thus volumes are areas and areas are spacings. Setting the spacing in the flat dimension makes this work, because rather than `v = dx * dy * dz` we have `v = dx * dy * 1` --- voila!

Reply:
Since tests are failing (gr....) we have some time to make these changes.

Reply:
Maybe just add some version of the text above?

Reply:
Damn the comment was already there and I didn't see it! That's OK!